src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import './App.scss';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilterStatus } from './helper';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoadedTodos, setIsLoadedTodos] = useState(false);
+  const [isTodoInfoRequested, setIsTodoInfoRequested] = useState(false);
+  const [filterCondition, setFilterCondition] = useState(FilterStatus.ALL);
+  const [query, setQuery] = useState('');
+  const [userId, setUserId] = useState<number | null>(null);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoadingError, setIsLoadingError] = useState(false);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const loadedTodos = await getTodos();
+
+        setIsLoadedTodos(true);
+        setTodos(loadedTodos);
+      } catch (error) {
+        setIsLoadedTodos(true);
+        setIsLoadingError(true);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      const queriedTodo = todo.title.toLowerCase().includes(query.toLowerCase());
```

Comment:
You repeat `query.toLowerCase()` this logic every iteration. Let's move it

```suggestion
  const filteredTodos = useMemo(() => {
    const preparedQuery = query.toLowerCase();
    
    return todos.filter(todo => {
      const queriedTodo = todo.title.toLowerCase().includes(preparedQuery));
```